plugins/src/main/kotlin/extension/AnvilExtensions.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package extension
+
+import org.gradle.api.Project
+import com.squareup.anvil.plugin.AnvilExtension
+import org.gradle.accessors.dm.LibrariesForLibs
+import org.gradle.kotlin.dsl.the
+
+/**
+ * Setup Anvil plugin with the given configuration.
+ * @param generateDaggerCode whether to enable general Dagger code generation using Kapt
+ * @param generateDaggerFactoriesUsingAnvil whether to generate Dagger factories using Anvil instead of Kapt
+ */
+fun Project.setupAnvil(
+    generateDaggerCode: Boolean = false,
+    generateDaggerFactoriesUsingAnvil: Boolean = true,
+) {
+    val libs = the<LibrariesForLibs>()
+    // Apply plugins and dependencies
+    applyPluginIfNeeded("com.squareup.anvil")
+
+    if (generateDaggerCode) {
+        applyPluginIfNeeded("org.jetbrains.kotlin.kapt")
```

Comment:
If you decide to do the change above in you'll need to change here too to

`applyPluginIfNeeded(libs.plugins.kapt)`